### PR TITLE
Tasklist new branch

### PR DESCRIPTION
**Pull request by DmytroArseniuk**

### CODE REVIEW

**Comment by natalia-zolochevska** [comment]
Great job! But I'd like to see more validation: custom validator and using of validation styles, not just properties. Please name pull request with some meanful description.

File: Tasks.iml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="WEB_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (natalia-zolochevska) [inline Tasks.iml]: don't commit iml files.

File: app/views/task-list.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        <a href="#/my">my</a>
+    </div>
+    <div ng-show="tasks.length == 0">Not tasks yet!</div>
+    <div class="Task" ng-repeat="task in tasks | filter: owner"
+         ng-mouseover="setEditableTask(task.id)"
+         ng-mouseleave="setEditableTask(null)">
+        <br>
+
+        <div ng-show="editableTaskId != task.id || task.isComplete"
+             ng-class="{true: 'CompletedTask', false: ''}[task.isComplete]">
```
Comment (natalia-zolochevska) [inline app/views/task-list.html]: discuss naming of boolean vars once again

File: app/views/task-list.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                  ng-blur="setFocus(null)"></textarea>
+
+        <input type="text"
+               ng-model="task.newAssignee"
+               ng-init="task.newAssignee = task.assignee.name"
+               placeholder="Assign to"
+               ng-show="editableTaskId == task.id && !task.isComplete">
+
+        <div class="Date">{{task.date | dateFilter}}</div>
+        <div class="TaskActions" ng-show="editableTaskId == task.id">
```
Comment (natalia-zolochevska) [inline app/views/task-list.html]: css classes - first letter lower case

File: .gitignore [hunk 4]
```diff
@@ -1,2 +1,3 @@
 /.idea/*
-TaskList.iml
\ No newline at end of file
+TaskList.iml
```
Comment (akobylianskiy) [inline .gitignore]: If you have project with multiple modules, you will be forced to ignore each 'iml' file with this approach. Generally I I use this rows in gitignore for IntelliJ projects to ignore all redundant files:
.idea/
*.iml
*.iws

File: app/js/scripts/app.js [hunk 5]
```diff
@@ -1,79 +1,24 @@
...
+            .state('tasks.my', {
+                url: "my",
+                templateUrl: "views/task-list.html"
+            })
+            .state('tasks.all', {
+                url: "all",
+                templateUrl: "views/task-list.html"
+            })
+    })
+    .filter('dateFilter', function ($filter) {
```
Comment (akobylianskiy) [inline app/js/scripts/app.js]: To my mind it's not a good idea to define your custom filter in app.js. You can use separate file for this.

File: app/js/scripts/controllers/tasks-front.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
+angular.module('tasks-front', ['tasks-back'])
```
Comment (akobylianskiy) [inline app/js/scripts/controllers/tasks-front.js]: You don't need to specify dependency for 'tasks-back' as you have already defined it in app.js.

File: app/js/scripts/controllers/tasks-front.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+            link: function (scope, elem, attr, ngModel) {
+
+                //For DOM -> model validation
+                ngModel.$parsers.unshift(function (value) {
+                    var valid = value.length > 0 && value.length < 15;
+                    ngModel.$setValidity('nameLength', valid);
+                    return valid ? value : undefined;
+                });
+            }
+        };
```
Comment (akobylianskiy) [inline app/js/scripts/controllers/tasks-front.js]: Please replace directive to separate js file.

Comment (natalia-zolochevska) [inline app/js/scripts/controllers/tasks-front.js]: Dmitry, use "extract". "Replace" is not a proper word here.

Natalia Zolochevska

Software Developer | Enterprise Technologies | TeamDev, Ltd.
Phone: +380 57 766-0163 x 2017

Natalia.Zolochevska@teamdev.com | www.teamdev.com

If you have been sent this email in error, please forward it to mismail @
teamdev.com and delete the original. Any opinions expressed within email
are those of the sender and do not necessarily represent the official
opinion of TeamDev Ltd.

File: app/views/task-list.html [hunk 8]
```diff
@@ -0,0 +1,43 @@
+<div class="TaskList" ng-controller="listItemController">
+
+    <div>Tasks for {{taskOwner}}<br>
+        <a href="#/all">all</a>,
+        <a href="#/my">my</a>
+    </div>
+    <div ng-show="tasks.length == 0">Not tasks yet!</div>
+    <div class="Task" ng-repeat="task in tasks | filter: owner"
+         ng-mouseover="setEditableTask(task.id)"
+         ng-mouseleave="setEditableTask(null)">
```
Comment (dkalinin) [inline app/views/task-list.html]: It would be better if task description editing was on click.

File: .gitignore (hunk 4 above)
Comment (DmytroArseniuk) [inline .gitignore]: Thank you for advice.

File: Tasks.iml (hunk 1 above)
Comment (DmytroArseniuk) [inline Tasks.iml]: Okay, I thought that it will resolve problem with importing project to IDEA.

File: app/views/task-list.html (hunk 8 above)
Comment (DmytroArseniuk) [inline app/views/task-list.html]: I agree, but it was in task description to edit task in this way.

File: app/js/scripts/app.js (hunk 5 above)
Comment (DmytroArseniuk) [inline app/js/scripts/app.js]: Good idea, but for this small application, probably, extracting filters and directives to another files will make reading of application more complex. But if I will have more than one directive and one filter I will extract them.

File: app/views/task-list.html (hunk 2 above)
Comment (DmytroArseniuk) [inline app/views/task-list.html]: Okay

File: app/views/task-list.html (hunk 3 above)
Comment (DmytroArseniuk) [inline app/views/task-list.html]: Okay

**Pull request merged**